From 105d4fdab464b6726caadb49818998ea81f1a012 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Fri, 14 Nov 2014 09:03:50 +0800
Subject: [PATCH] armada38x: Fix the array bounds warning

It use the implicitly defined for the array, and use MV_CONFIG_TYPE_MAX_OPTION
to access the array. If the size of the array is less than
MV_CONFIG_TYPE_MAX_OPTION, The warning will generate.

The define and using code as the following:
MV_BOARD_CONFIG_TYPE_INFO boardConfigTypesInfo[] = MV_BOARD_CONFIG_INFO;

for (i = 0; i < MV_CONFIG_TYPE_MAX_OPTION ; i++)
	if (boardConfigTypesInfo[i].configId == configClass) {
		*configInfo = boardConfigTypesInfo[i];
		if (boardConfigTypesInfo[i].isActiveForBoard[boardId])
			return MV_TRUE;
		else
			return MV_FALSE;
	}

The generating warning:
arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c:
In function 'mvBoardConfigTypeGet':
arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c:1868:27:
warning: array subscript is above array bounds [-Warray-bounds]

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../armada_38x_family/boardEnv/mvBoardEnvLib.c     |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
index e22ddb0..2673233 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
@@ -92,7 +92,7 @@
 
 extern MV_BOARD_INFO *marvellBoardInfoTbl[];
 extern MV_BOARD_INFO *customerBoardInfoTbl[];
-MV_BOARD_CONFIG_TYPE_INFO boardConfigTypesInfo[] = MV_BOARD_CONFIG_INFO;
+MV_BOARD_CONFIG_TYPE_INFO boardConfigTypesInfo[MV_CONFIG_TYPE_MAX_OPTION] = MV_BOARD_CONFIG_INFO;
 MV_BOARD_SATR_INFO boardSatrInfo[] = MV_SAR_INFO;
 MV_SATR_BOOT_TABLE satrBootSrcTable[] = MV_SATR_BOOT_SRC_TABLE_VAL;
 
-- 
1.7.5.4

